### PR TITLE
Fix null case bug in BannedUserPage and properly add no expiration text

### PR DESCRIPTION
**Pull request by sudokoko**
This PR fixes the issue where the ban information page would improperly pull the user's latest moderation case, pulling the latest dismissed case rather than the latest case where expiration isn't null (which should theoretically never happen anyways, but should still check since it is marked as nullable in the database).

Also properly implements the existing permanent ban string into the information page, as well as fixes a missing page element when the case is null.

### CODE REVIEW

File: ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml [hunk 1]
```diff
@@ -20,7 +22,7 @@
         @if (Model.ModCase != null)
         {
             <span>
-                @if (Model.ModCase.ExpiresAt != null)
+                @if (Model.ModCase.ExpiresAt < maximumExpiration)
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml:25]:
```suggestion
                @if (Model.ModCase.ExpiresAt < DateTime.MaxValue)
```

File: ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml [hunk 2]
```diff
@@ -9,6 +9,8 @@
 
     string timeZone = (string?)ViewData["TimeZone"] ?? TimeZoneInfo.Local.Id;
     TimeZoneInfo timeZoneInfo = TimeZoneInfo.FindSystemTimeZoneById(timeZone);
+
+    DateTime maximumExpiration = new(9999, 12, 31, 23, 59, 00, DateTimeKind.Utc);
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml:13]:
```suggestion
```

File: ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml [hunk 3]
```diff
@@ -30,6 +32,12 @@
...
             </span>    
         }
+        else
+        {
+            <span>
+                @Model.Translate(ModerationStrings.SuspensionExpiration, Model.Translate(GeneralStrings.Unknown))
+            </span>
+        }
 
         <ul>
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml:42]: According to HTML specs and Rider analysis, you shouldn't be putting a list inside a paragraph object, you can either move the list outside the paragraph tag or turn it into a div

File: ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml [hunk 4]
```diff
@@ -30,6 +32,12 @@
...
             <li>@Model.Translate(ModerationStrings.LbpOnlineMultiplayer)</li>
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml:43]: GitHub doesn't let me comment on other lines but on line 53 there is no handler for if a reason isn't given, it just looks like `Reason for suspension: ""`

File: ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml [hunk 5]
```diff
@@ -20,7 +22,7 @@
         @if (Model.ModCase != null)
         {
             <span>
-                @if (Model.ModCase.ExpiresAt != null)
+                @if (Model.ModCase.ExpiresAt < maximumExpiration)
                 {
                     @Model.Translate(ModerationStrings.SuspensionExpiration, TimeZoneInfo.ConvertTime(Model.ModCase.ExpiresAt.Value, timeZoneInfo).ToString("M/d/yyyy @ h:mm tt"))
                 }
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml:28]: On line 31, the wording sounds odd in the full sentence:
```
During this suspension, the following features will be limited until Does not expire:
```

**Review by Slendy — CHANGES_REQUESTED** [review]

File: ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml (hunk 2 above)
Comment (sudokoko) [reply ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml:13]: This is intentional. The MaxValue expression actually exceeds the maximum value the form lets you submit.

Comment (Slendy) [reply ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml:13]: Is it possible to make this field static then so it isn't being recreated for every request?

Comment (sudokoko) [reply ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml:13]: Will do that once I become available

File: ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml.cs [hunk 6]
```diff
@@ -34,7 +34,8 @@ public async Task<IActionResult> OnGet()
         if (user == null) return this.Redirect("~/login");
         if (!user.IsBanned) return this.Redirect("~/");
 
-        ModerationCaseEntity? modCase = await this.Database.Cases.OrderByDescending(c => c.CreatedAt)
+        ModerationCaseEntity? modCase = await this.Database.Cases
+			.OrderByDescending(c => c.CreatedAt)
```
Comment (sudokoko) [inline ProjectLighthouse.Servers.Website/Pages/Login/BannedUserPage.cshtml.cs:38]: Will fix this momentarily, Rider being weird.

**Review by Slendy — APPROVED** [review]
image

**Review by daggintosh — APPROVED** [review]

**Pull request merged**